src/driver/linux_onload/ossock_calls.c

```diff
@@ -94,10 +94,8 @@ oo_fd_replace_file(struct file* old_filp, struct file* new_filp,
     unsigned flags;
 
     task_unlock(current);
-    rcu_read_lock(); /* for files_fdtable() */
```

Comment:
I think you might have been misled by the recent appearance of 657639ded5 , which fixed exactly the same warning. The code you are modifying no longer calls `get_unused_fd_flags()`/`alloc_fd()` with RCU locked so cannot have caused the backtrace you saw.

Per the definition of `files_fdtable()` (and the lockdep verification therein) I believe we do still need the lock around the code that has it.